src/MOI_wrapper.jl

```diff
@@ -1600,46 +1636,81 @@ function MOI.get(
     return model.solution.rowvalue[row(model, c)+1]
 end
 
-function _dual_multiplier(model::Optimizer)
-    return MOI.get(model, MOI.ObjectiveSense()) == MOI.MAX_SENSE ? -1 : 1
-end
+"""
+    _sense_corrector(model::Optimizer)
 
+Return `-1` if `MAX_SENSE`, and `1` is `MIN_SENSE`. Useful for correcting
+solution attributes which are sense-dependent.
 """
-    _signed_dual(model::Optimizer, dual::Float64, ::Type{Set})
+function _sense_corrector(model::Optimizer)
+    senseP = Ref{Cint}()
+    ret = Highs_getObjectiveSense(model, senseP)
+    _check_ret(ret)
+    return senseP[]
+end
 
-A heuristic for determining whether the dual of an interval constraint applies
-to the lower or upper bound. It can be wrong by at most the solver's tolerance.
 """
-function _signed_dual end
+    _farkas_variable_dual(model::Optimizer, col::Cint)
 
-function _signed_dual(
-    model::Optimizer,
-    dual::Float64,
-    ::Type{MOI.LessThan{Float64}},
-)
-    return min(_dual_multiplier(model) * dual, 0.0)
-end
+Return a Farkas dual associated with the variable bounds of `col`. Given a dual
+ray:
 
-function _signed_dual(
-    model::Optimizer,
-    dual::Float64,
-    ::Type{MOI.GreaterThan{Float64}},
-)
-    return max(_dual_multiplier(model) * dual, 0.0)
-end
+    ā * x = λ' * A * x <= λ' * b = -β + sum(āᵢ * Uᵢ | āᵢ < 0) + sum(āᵢ * Lᵢ | āᵢ > 0)
 
-function _signed_dual(model::Optimizer, dual::Float64, ::Any)
-    return _dual_multiplier(model) * dual
+The Farkas dual of the variable is ā, and it applies to the upper bound if ā < 0,
+and it applies to the lower bound if ā > 0.
+"""
+function _farkas_variable_dual(model::Optimizer, col::Cint)
+    num_nz, num_cols = Ref{Cint}(0), Ref{Cint}(0)
+    # TODO(odow): how does getColsByRangeWork???
```

Comment:
Here's upstream: https://github.com/ERGO-Code/HiGHS/blob/2b89bd90391bb60edf252e0cbc7ea40f59798cf9/src/simplex/HighsSimplexInterface.cpp#L513-L573

It'd be nice to have the Gurobi/CPLEX behavior where you can pass `NULL` and have it populate `num_nz` so you can allocate the correct length.

Reply:
https://github.com/ERGO-Code/HiGHS/issues/465